src/environment.cpp

```diff
@@ -1589,9 +1589,29 @@ u32 ServerEnvironment::addParticleSpawner(float exptime)
 	return id;
 }
 
-void ServerEnvironment::deleteParticleSpawner(u32 id)
+u32 ServerEnvironment::addParticleSpawner(float exptime, u16 attached_id)
+{
+	u32 id = addParticleSpawner(exptime);
+	m_particle_spawner_attachments[id] = attached_id;
+	ServerActiveObject *obj = getActiveObject(attached_id);
+	if (obj != NULL) {
```

Comment:
You just said to use iterator methods inline since the rest of the code does it, but in this file the code typically uses != NULL.